upstairs/src/pseudo_file.rs

```diff
@@ -221,6 +210,16 @@ impl<T: BlockIO> CruciblePseudoFile<T> {
 
         Ok(())
     }
+    pub async fn activate_with_gen(
+        &self,
+        gen: u64,
+    ) -> Result<(), CrucibleError> {
+        self.block_io.activate_with_gen(gen).await
```

Comment:
CruciblePseudoFile needs to do some queries to set some of its fields after activation, those need to be duplicated here.